cmp/form/dropdown/BaseComboField.js

```diff
@@ -31,20 +36,18 @@ export class BaseComboField extends BaseDropdownField {
 
     forceSelectionToOptionOrRevert() {
         const {internalOptions, internalValue} = this;
+        if (!internalValue) return;
 
-        // 0) We have a match, nothing to be done
-        if (find(internalOptions, {label: internalValue})) {
```

Comment:
Removing this step. If our option labels differ from our values, our intervalValue might be a label match here, but we don't want that to remain the value of the field. We'd want to noteValueChanged(match.value) as below.